File: tests/metrics.test.js
const request = require('supertest');
const app = require('../app'); // Adjust path accordingly

describe('GET /metrics', () => {
    it('should return request counts per route', async () => {
        const response = await request(app).get('/metrics');
        expect(response.statusCode).toBe(200);
        expect(response.body).toHaveProperty('routes'); // Ensure structure is as expected
        // Add assertions for specific counts if needed
    });
});

File: tests/metrics.test.py
import requests
import pytest

BASE_URL = 'http://localhost:5000'  # Update accordingly


def test_metrics():
    # Making a sample request to increment count
    requests.post(f'{BASE_URL}/sample-route')
    response = requests.get(f'{BASE_URL}/metrics')
    assert response.status_code == 200
    data = response.json()
    assert 'routes' in data
    assert data['routes']['sample-route'] == 1  # Ensure the count is correct